src/org/opendatakit/briefcase/reused/OptionalProduct.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendatakit.briefcase.reused;
+
+import java.util.Optional;
+
+/**
+ * This class represents the type-safe Application of an arbitrary amount of {@link Optional} values.
```

Comment:
For discrete measurements, use number instead of amount. What amount of milk do you want? How many games did you win.

Reply:
Thanks for the explanation :)